### DISCUSSION TITLE
Ratelimit instead of dying with extremely verbose error?

### DISCUSSION THREAD

**Discussion by ghost** [nextcord/nextcord#1148] [General]
![image](https://github.com/nextcord/nextcord/assets/118899053/b3b7ee56-dd26-419b-a25f-02516999e958)
Could it manage to retry and wait long?

**Comment by ghost** [comment]
Using 2.5 version. Also how long to ratelimit to go away? Cloudflare bans needs 10000 requests but i didnt make that much. I think i made some 404 but not sure how much 404 before getting rate limit

**Reply by Skelmis** [reply]
A cloudflare ban is not worth waiting out and is far more serious then a 429. If I can recall correctly that is a day ban.

It can also happen if your using shared hosting for your bot which uses a shared IP.

**Reply by ghost** [reply]
@Skelmis I have a server where its collectoin of other server but in that channel links are expired. So I was doing fetch_invite for all link, if link is valid it stores in database and if invalid it deletes that message. Cloudflare is triggered only by fetch_invite so I assume we shouldnt be using fetch_invites a lot or maybe its due to lots of invalid invites in that channel? Can rerunning it lead to permanent ban?

**Reply by Skelmis** [reply]
There is a lot of context missing here which would plausibly be better suited to our help server if you have access?

Otherwise a cloudflare ban generally dictates you've done a excessive amount of requests. How do you run your bot?
